be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala

```diff
@@ -259,9 +261,48 @@ sealed class ElectionValidator(dbActorRef: => AskableActorRef) extends MessageDa
     }
   }
 
-// TODO: Proper validation
   def validateResultElection(rpcMessage: JsonRpcRequest): GraphMessage = {
-    Right(rpcMessage)
+    def validationError(reason: String): PipelineError = super.validationError(reason, "ResultElection", rpcMessage.id)
+    rpcMessage.getParamsMessage match {
+      case Some(_: Message) =>
+        val (resultElection, _, _, channel) = extractData[ResultElection](rpcMessage)
+        val electionQuestionResult = resultElection.questions
+        val electionId = channel.extractChildChannel
+        val electionQuestions = extractElectionQuestions(channel, dbActorRef)
+        if (!electionQuestions.isDefined) {
+          Left(validationErrorNoMessage(rpcMessage.id))
+        }
+        runChecks(
+          checkNumberOfVotes(
+            rpcMessage,
+            channel,
+            electionQuestionResult,
+            dbActorRef,
+            validationError(s"trying to send a ResultElection message with an invalid number of votes.")
+          ),
+          checkElectionEnded(
+            rpcMessage,
+            channel,
+            dbActorRef,
+            validationError("trying to send a ResultElection while Election is not ended yet.")
+          ),
+          checkQuestionId(
```

Comment:
Not sure why this check is needed here. Here you seem to verify the question ids of the Setup election message. But this is already done in the validator of SetupElection. Thus, you can remove this check.